lib/elixir/lib/kernel/utils.ex

```diff
@@ -29,9 +29,16 @@ defmodule Kernel.Utils do
     append_first? = Keyword.get(opts, :append_first, false)
 
     {name, args} =
-      case Macro.decompose_call(fun) do
-        {_, _} = pair -> pair
-        _ -> raise ArgumentError, "invalid syntax in defdelegate #{Macro.to_string(fun)}"
+      case fun do
+        {:when, _, [_left, right]} ->
```

Comment:
```suggestion
        {:when, _, [_left, [right]]} ->
```

Otherwise I think Macro.to_string will  print the square brackets

Reply:
When is a binary operator, it has only left and right sides. The code snippets are also equivalent:

```
iex(1)> [1 | [2]]
[1, 2]
```

Reply:
> When is a binary operator, it has only left and right sides. The code snippets are also equivalent:
> 
> ```
> iex(1)> [1 | [2]]
> [1, 2]
> ```

oops. I overlooked the comma
Nice trick